pkg/reconciler/pipeline/controller.go

```diff
@@ -830,39 +830,45 @@ func (r *pipelineReconciler) resumePipeline(ctx context.Context, pl *dfv1.Pipeli
 }
 
 func (r *pipelineReconciler) pausePipeline(ctx context.Context, pl *dfv1.Pipeline) (bool, error) {
+	var (
+		drainCompleted = false
+		daemonClient   daemonclient.DaemonClient
+		err            error
+	)
 	// check that annotations / pause timestamp annotation exist
 	if pl.GetAnnotations() == nil || pl.GetAnnotations()[dfv1.KeyPauseTimestamp] == "" {
 		patchJson := `{"metadata":{"annotations":{"` + dfv1.KeyPauseTimestamp + `":"` + time.Now().Format(time.RFC3339) + `"}}}`
 		if err := r.client.Patch(ctx, pl, client.RawPatch(types.MergePatchType, []byte(patchJson))); err != nil && !apierrors.IsNotFound(err) {
 			return true, err
 		}
+		pl.Status.MarkPhasePausing()
+		updated, err := r.scaleDownSourceVertices(ctx, pl)
```

Comment:
If scaling down source fails, but annotating succeeds, then it will not be able to enter this secion.